pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -2163,6 +2163,28 @@ func (a *ApplicationBackupController) cleanupResources(
 		log.ApplicationBackupLog(backup).Errorf("%v", errMsg)
 		return err
 	}
+	// Check if the backup is of the type virtual machine and then collect the rules and only then delete it.
+	if IsBackupObjecyTypeVirtualMachine(backup) {
+		storkVmRules, err := storkops.Instance().ListRules(backup.Namespace, metav1.ListOptions{})
```

Comment:
Realised that for schedules there may exist many rules. Instead of fetching all rules in the namespace. We could fetch only the rule tagged in backup.Spec.preExecRule  and backup.Spec.postExecRule. This way we dont need to fetch rules that are not referred in this backup.